pkg/spdk/engine.go

```diff
@@ -353,7 +353,13 @@ func (e *Engine) ValidateAndUpdate(
 		if err != nil {
 			return err
 		}
+
 		if err := initiator.LoadNVMeDeviceInfo(); err != nil {
+			if strings.Contains(err.Error(), "connecting state") ||
```

Comment:
nit: determining different errors by parsing string would be potentially error-prone. I would suggest having gRPC error like each error still implements the error interface but also has a specific interface to provide error code capability. 

This can be a way to improve our error handling, especially need to differentiate different error types.

Reply:
Personally, I like to use error code. However, it seems there is no good error code lib we can use directly.
We may can implement an error interface lib for our applications.

Reply:
Sounds good. Let's create an issue for this improvement later.

Reply:
Tracked in https://github.com/longhorn/longhorn/issues/6093